File: backend/app/models/agent.py
import uuid
from datetime import datetime, timezone
import enum

from sqlalchemy import String, Integer, DateTime, ForeignKey, Enum as SAEnum, JSON
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.core.database import Base


class AgentStatus(str, enum.Enum):
    creating = "creating"
    running = "running"
    stopped = "stopped"
    error = "error"


class Agent(Base):
    __tablename__ = "agents"

    id: Mapped[str] = mapped_column(String(36), primary_key=True, default=lambda: str(uuid.uuid4()))
    user_id: Mapped[str] = mapped_column(String(36), ForeignKey("users.id", ondelete="CASCADE"), index=True)
    name: Mapped[str] = mapped_column(String(100))
    model_name: Mapped[str] = mapped_column(String(50), default="deepseek-chat")
    container_id: Mapped[str | None] = mapped_column(String(100), nullable=True)
    gateway_token: Mapped[str | None] = mapped_column(String(200), nullable=True)
    status: Mapped[AgentStatus] = mapped_column(SAEnum(AgentStatus), default=AgentStatus.creating)
    config: Mapped[dict | None] = mapped_column(JSON, nullable=True)
    ws_port: Mapped[int | None] = mapped_column(Integer, nullable=True)
    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), default=lambda: datetime.now(timezone.utc)
    )
    last_active_at: Mapped[datetime | None] = mapped_column(DateTime(timezone=True), nullable=True)

    user = relationship("User", back_populates="agents")
    conversations = relationship("Conversation", back_populates="agent", cascade="all, delete-orphan")

File: ios/Clawzy/clawzy/Models/Agent.swift
import Foundation

struct Agent: Codable, Identifiable {
    let id: String
    let userId: String?
    let name: String
    let modelName: String
    let containerId: String?
    let status: AgentStatus
    let wsPort: Int?
    let createdAt: String
    let lastActiveAt: String?

    enum CodingKeys: String, CodingKey {
        case id
        case userId = "user_id"
        case name
        case modelName = "model_name"
        case containerId = "container_id"
        case status
        case wsPort = "ws_port"
        case createdAt = "created_at"
        case lastActiveAt = "last_active_at"
    }
}

enum AgentStatus: String, Codable {
    case creating
    case running
    case stopped
    case error
}

struct CreateAgentRequest: Codable {
    let name: String
    let modelName: String

    enum CodingKeys: String, CodingKey {
        case name
        case modelName = "model_name"
    }
}
